### ISSUE TITLE
why gdpp loss always (almost) negative， it can't compute BCE loss

### ISSUE THREAD

**Issue by Johnson-yue** [M-Elfeki/GDPP#2]
Hi, I was trying your gdpp loss with pytorch version,  but when I add gdpp loss on G_loss,  the training process crash, because the G_loss(~1.0) + gdpp_loss is negative. So, it can't compute BCE loss  .

My question is  why is gdpp loss always negative, does it make sense??

**Comment by HelenMao** [comment]
I am trying gdpp loss with PyTorch version too, it seems that the problem lies in that the PyTorch has no implementation of the backward of torch.eig which calculates the eigenvalues and eigenvectors.

**Comment by Johnson-yue** [comment]
@HelenMao Hi，I read your paper《MSGAN》， happy to see you， Do you mean torch only implementation forward for torch.eig but not implement of backward??

**Comment by HelenMao** [comment]
Right, I am trying to use the given PyTorch code, but I find the torch.eig has not been implemented the backward. Only the torch.symeig has the implementation of backward, but it can not be directly used.
Also, I am running the TensorFlow codes of synthetic data, but it can not reproduce the results in the paper. I get the zero-mode when running the models, it is really strange.

**Comment by Johnson-yue** [comment]
yes， When GDPP have released ， I test many experiment but failed，So， I did not followed it。
good luck

**Comment by HelenMao** [comment]
Thanks for sharing the information.

**Comment by schuy1er** [comment]
> Thanks for sharing the information.

So have you reproduced the results in the paper? I‘m trying to reproduce the gdpp on cifar but i cannot reproduce such a high inception score.
